binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java

```diff
@@ -64,8 +64,13 @@ public void initialize() {
         Configuration config = getThing().getConfiguration();
 
         location = (String) config.get("location");
-        if ("us".equalsIgnoreCase((String) config.get("unit"))) {
+        if ("metric".equalsIgnoreCase((String) config.get("unit"))) {
+            unit = "c";
+        } else if ("us".equalsIgnoreCase((String) config.get("unit"))) {
             unit = "f";
+        } else {
+            // means unit not known
+            unit = null;
```

Comment:
I am not sure if this is the right thing to do. Imho we should either
a) cancel the initialization as there is an invalid configuration
b) ignore an invalid unit parameter and go for the default, which is "metric" as per the thing type definition. But the implementation now uses the default of Yahoo weather, which is fahrenheit.